### DISCUSSION TITLE
Messenger and large table

### DISCUSSION THREAD

**Discussion by nitrobear** [symfony/symfony#47291] [Q&A] [answered]
I have my table messenger_messages with 24'000'000 records.

When I try to process, the query take more than 4 minutes... 

by exemple : 
`mysql> SELECT m.* FROM messenger_messages m WHERE (m.delivered_at is null OR m.delivered_at < "2022-08-16 13:25:13") AND (m.available_at <= "2022-08-16 13:25:13") AND (m.queue_name = "default") ORDER BY available_at ASC LIMIT 1 ;`

give : 
`
1 row in set (4 min 45.60 sec)
`
 
How can I consume all this message in a correct time ?

**Comment by lyrixx** [comment, marked as answer]
Hello, Is the a table that contains failed message or regular message?

If it's failed: You should have monitored sonner this table, and retry / delete theses message much sonner. In our application, we have an alter if this tables contains more than 10 messages

If it's regular: something is wrong here. Messenger is done for processing (almost) real time data. There is some use case where it's possible to end up with 2.4M messages, but it's rare. And in that case, it's better to use a real storage engine (like rabbitmq)

**Reply by nitrobear** [reply]
The table have regular message.
I don't know why, but we have a problem one day, and messenger store a lot of message, and know, it take to much time for process it.
I will move it to rabbitmq.
